src/social_db/data/set/call_function_args_type.rs

```diff
@@ -0,0 +1,105 @@
+use std::str::FromStr;
+
+use color_eyre::eyre::Context;
+use inquire::Select;
+use strum::{EnumDiscriminants, EnumIter, EnumMessage, IntoEnumIterator};
+
+#[derive(Debug, EnumDiscriminants, Clone, clap::ValueEnum)]
+#[strum_discriminants(derive(EnumMessage, EnumIter))]
+/// How do you want to pass the function call arguments?
+pub enum FunctionArgsType {
+    #[strum_discriminants(strum(
+        message = "with-json        - Valid JSON arguments (e.g. {\"token_id\": \"42\"})"
+    ))]
+    /// Valid JSON arguments (e.g. {"token_id": "42"})
+    WithJson,
+    #[strum_discriminants(strum(message = "text-args        - Arbitrary text arguments"))]
+    /// Arbitrary text arguments
+    WithText,
+    #[strum_discriminants(strum(
+        message = "with-json-file   - Reading from a reusable text file"
+    ))]
+    /// Reading from a reusable text file
+    WithJsonFile,
+    #[strum_discriminants(strum(
+        message = "with-text-file   - Reading from a reusable JSON file"
+    ))]
+    /// Reading from a reusable JSON file
+    WithTextFile,
+}
+
+impl interactive_clap::ToCli for FunctionArgsType {
+    type CliVariant = FunctionArgsType;
+}
+
+impl std::str::FromStr for FunctionArgsType {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "with-json" => Ok(Self::WithJson),
+            "with-text" => Ok(Self::WithText),
+            "with-json-file" => Ok(Self::WithJsonFile),
+            "with-text-file" => Ok(Self::WithTextFile),
+            _ => Err("FunctionArgsType: incorrect value entered".to_string()),
+        }
+    }
+}
+
+impl std::fmt::Display for FunctionArgsType {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::WithJson => write!(f, "with-json"),
+            Self::WithText => write!(f, "with-text"),
+            Self::WithJsonFile => write!(f, "with-json-file"),
+            Self::WithTextFile => write!(f, "with-text-file"),
+        }
+    }
+}
+
+impl std::fmt::Display for FunctionArgsTypeDiscriminants {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::WithJson => write!(f, "with-json"),
+            Self::WithText => write!(f, "with-text"),
+            Self::WithJsonFile => write!(f, "with-json-file"),
+            Self::WithTextFile => write!(f, "with-text-file"),
```

Comment:
It would be better to use subcommands instead of this ugly hack (the same goes for near-cli-rs where we use this same approach now). Is there any reason we cannot use subcommand?

Reply:
Done